content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,274 @@
+
+Now, let's start coding. In the `main.py` file that *pycharm* automatically creates for you, type in the following code:
+
+```python
+import cv2
+import mediapipe as mp
+from math import hypot
+from ctypes import cast, POINTER
+from comtypes import CLSCTX_ALL
+from pycaw.pycaw import AudioUtilities, IAudioEndpointVolume
+import numpy as np
+```
+In the above code segment, we import each library we installed in our project.
+
+```python
+cap = cv2.VideoCapture(0) 
+```
+We are getting the video input from our computer's primary camera. If you are using any other camera, replace the number `0` with that of the camera you are using.
+
+### Detecting, initializing, and configuring the hands
+```python
+mpHands = mp.solutions.hands
+hands = mpHands.Hands()
+mpDraw = mp.solutions.drawing_utils
+```
+`mp.solutions.hands`: we are calling on the `mediapipe` hand module to detect the hands from the video input we got from our primary camera.
+
+`mpHands.Hands()`: this completes the initialization and configuration of the hands that the mediapipe hand module detected.
+
+`mp.solutions.drawing_utils`: draws the *connections* and *landmarks* on the hand that mediapipe hand module detected. 
+
+### Accessing the speaker using pycaw
+```python
+devices = AudioUtilities.GetSpeakers()
+interface = devices.Activate(IAudioEndpointVolume._iid_, CLSCTX_ALL, None)
+volume = cast(interface, POINTER(IAudioEndpointVolume))
+```
+These are the initializations we need for `pycaw`. The developer provides this library together with the initializations. We are not going to change anything. You can find the documentation [here](https://github.com/AndreMiras/pycaw).
+
+### Finding the volume range between the minimum and maximum volume
+```python
+volMin, volMax = volume.GetVolumeRange()[:2]
+```
+We are calling the `pycaw` `volume.GetVolumeRange()` function to find the volume range between the minimum and maximum volume.
+
+### Capturing an image from our camera and converting it to an RGB image
+```python
+while True:
+    success, img = cap.read()
+    imgRGB = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+    results = hands.process(imgRGB)
+```
+`success, img = cap.read()`: checks whether the camera we have specified works. If it works, we will capture an image.
+
+`imgRGB = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)`: converts the image captured by the camera to an `RGB` image.
+
+` results = hands.process(imgRGB)`: completes the processing of the image that was converted to `RGB`.
+
+### Checking whether we have multiple hands in our input
+```python
+lmList = []
+if results.multi_hand_landmarks: 
+```
+`lmList = []`: this code creates an *empty list* which will store the list of elements of the hands detected by the `mediapipe` hand module, i.e., the number of points on the hand.
+
+`if results.multi_hand_landmarks:` checks whether the input has multiple hands.
+
+### Creating a for loop to manipulate each hand

```

Comment:
There is an issue with how you are explaining your code. In the prerequisites, you specified that knowledge in python is required. So I don't find it necessary to repeat each and every line of code and what it does. Since the readers know about syntax and control structures, you can give an overview of what the code does and what the output may be in terms of short paragraphs. The same way you have done in the `Finding the distance between points 4 and 8` and `Terminating the program` sections. 

Again, go through published tutorials to see how they go about article writing.